src/ServiceStack/HttpRequestExtensions.cs

```diff
@@ -1111,5 +1111,19 @@ public static string GetRequestValue(this IHttpRequest req, string name)
                     throw new NotSupportedException($"Unknown IHttpRequest property '{name}'");
             }
         }
+
+        public static void EachRequest<T>(this IRequest httpReq, Action<T> action)
+        {
+            if (!(httpReq.Dto is IEnumerable<T> requests))
+                return;
+
+            requests.Each((i, dto) =>
+            {
+                httpReq.Items["AutoBatchIndex"] = i;
+                action(dto);
+            });
+
```

Comment:
Ideally you should avoid LINQ in library code for reduced allocations, e.g. using `foreach` or `for` .